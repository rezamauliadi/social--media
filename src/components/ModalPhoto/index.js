import React from "react";
import PropTypes from "prop-types";

import { Button, Image, Modal } from "semantic-ui-react";

const ModalPhoto = ({ photo, showPhoto, onModalClose }) => {
  return (
    <Modal open={showPhoto} onClose={onModalClose} closeIcon>
      <Modal.Header>Title: {photo.title}</Modal.Header>
      <Modal.Content>
        <center>
          <Image src={photo.url} style={{ zIndex: "1001" }} />
        </center>
      </Modal.Content>
      <Modal.Actions>
        <Button positive content="Nice!" onClick={onModalClose} />
      </Modal.Actions>
    </Modal>
  );
};

ModalPhoto.propTypes = {
  photo: PropTypes.object
};

export default ModalPhoto;
