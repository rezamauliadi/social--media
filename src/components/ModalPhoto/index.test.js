import React from "react";
import { shallow } from "enzyme";

import ModalPhoto from "./index.js";

describe("Component: ModalPhoto", () => {
  const photo = {
    albumId: 1,
    id: 1,
    title: "accusamus beatae ad facilis cum similique qui sunt",
    url: "https://via.placeholder.com/600/92c952",
    thumbnailUrl: "https://via.placeholder.com/150/92c952"
  };

  const defaultProps = {
    photo,
    onModalClose: jest.fn()
  };

  it("should match snapshot when show photo is true", () => {
    const props = { ...defaultProps, showPhoto: true };
    const component = shallow(<ModalPhoto {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should match snapshot when show photo is false", () => {
    const props = { ...defaultProps, showPhoto: false };
    const component = shallow(<ModalPhoto {...props} />);

    expect(component).toMatchSnapshot();
  });
});
