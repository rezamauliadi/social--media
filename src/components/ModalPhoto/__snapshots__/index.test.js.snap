// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ModalPhoto should match snapshot when show photo is false 1`] = `
<Modal
  centered={true}
  closeIcon={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[MockFunction]}
  open={false}
>
  <ModalHeader>
    Title: 
    accusamus beatae ad facilis cum similique qui sunt
  </ModalHeader>
  <ModalContent>
    <center>
      <Image
        as="img"
        src="https://via.placeholder.com/600/92c952"
        style={
          Object {
            "zIndex": "1001",
          }
        }
        ui={true}
      />
    </center>
  </ModalContent>
  <ModalActions>
    <Button
      as="button"
      content="Nice!"
      onClick={[MockFunction]}
      positive={true}
    />
  </ModalActions>
</Modal>
`;

exports[`Component: ModalPhoto should match snapshot when show photo is true 1`] = `
<Modal
  centered={true}
  closeIcon={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[MockFunction]}
  open={true}
>
  <ModalHeader>
    Title: 
    accusamus beatae ad facilis cum similique qui sunt
  </ModalHeader>
  <ModalContent>
    <center>
      <Image
        as="img"
        src="https://via.placeholder.com/600/92c952"
        style={
          Object {
            "zIndex": "1001",
          }
        }
        ui={true}
      />
    </center>
  </ModalContent>
  <ModalActions>
    <Button
      as="button"
      content="Nice!"
      onClick={[MockFunction]}
      positive={true}
    />
  </ModalActions>
</Modal>
`;
